File: src/java/eu/sig/training/ch02/Employees.java
package eu.sig.training.ch02;

public class Employees {
    // tag::printDepartmentEmployees[]    
    public static void printDepartmentEmployees(String department) {
        Query q = new Query();
        for (Employee e : q.addColumn("FamilyName")
            .addColumn("Initials")
            .addColumn("GivenName")
            .addColumn("AddressLine1")
            .addColumn("ZIPcode")
            .addColumn("City")
            .addTable("EMPLOYEES")
            .addWhere("EmployeeDep='" + department + "'")
            .execute()) {
            System.out.println("<div name='addressDiv'" + e.getFamilyName()
                + ", " + e.getInitials() + "<br />" + e.getAddressLine1()
                + "<br />" + e.getZipCode() + e.getCity() + "</div>");
        }
    }
    // end::printDepartmentEmployees[]
}

class Query {

    @SuppressWarnings("unused")
    public Query addColumn(String string) {
        return null;
    }

    public Employee[] execute() {
        return null;
    }

    @SuppressWarnings("unused")
    public Query addWhere(String string) {
        return null;
    }

    @SuppressWarnings("unused")
    public Query addTable(String string) {
        return null;
    }
}

class Employee {

    public String getFamilyName() {
        return null;
    }

    public String getAddressLine1() {
        return null;
    }

    public String getZipCode() {
        return null;
    }

    public String getCity() {
        return null;
    }

    public String getInitials() {
        return null;
    }
}

File: src/csharp/eu/sig/training/ch02/Employees.cs
using System;

namespace eu.sig.training.ch02
{
    public static class Employees
    {
        // tag::printDepartmentEmployees[]
        public static void PrintDepartmentEmployees(string department)
        {
            Query q = new Query();
            foreach (Employee e in q.AddColumn("FamilyName")
                .AddColumn("Initials")
                .AddColumn("GivenName")
                .AddColumn("AddressLine1")
                .AddColumn("ZIPcode")
                .AddColumn("City")
                .AddTable("EMPLOYEES")
                .AddWhere($"EmployeeDep='{department}'")
                .Execute())
            {
                Console.WriteLine($@"<div name='addressDiv'>
                    {e.FamilyName}, {e.Initials}<br />" +
                    "{e.AddressLine1}<br />{e.ZipCode}{e.City}</div>");
            }
        }
        // end::printDepartmentEmployees[]
    }

    class Query
    {

        public Query AddColumn(string s)
        {
            return null;
        }

        public Employee[] Execute()
        {
            return null;
        }

        public Query AddWhere(string s)
        {
            return null;
        }

        public Query AddTable(string s)
        {
            return null;
        }
    }

    class Employee
    {

        public string FamilyName { get; set; }

        public string AddressLine1 { get; set; }

        public string ZipCode { get; set; }

        public string City { get; set; }

        public string Initials { get; set; }
    }
}
